File: app/src/main/java/gohryt/sapphire/resources/Colors.java
package gohryt.sapphire.resources;

import android.content.Context;
import android.os.Build;

import androidx.annotation.RequiresApi;
import androidx.core.content.ContextCompat;

import gohryt.sapphire.R;

@RequiresApi(Build.VERSION_CODES.R)
public class Colors {
    public int foregroundMain;
    public int foregroundPoor;
    public int foregroundAccent;

    public int backgroundMain;
    public int backgroundActive;
    public int backgroundPoor;

    public Colors(Context context) {
        foregroundMain = ContextCompat.getColor(context, R.color.foregroundMain);
        foregroundPoor = ContextCompat.getColor(context, R.color.foregroundPoor);
        foregroundAccent = ContextCompat.getColor(context, R.color.foregroundAccent);

        backgroundMain = ContextCompat.getColor(context, R.color.backgroundMain);
        backgroundActive = ContextCompat.getColor(context, R.color.backgroundActive);
        backgroundPoor = ContextCompat.getColor(context, R.color.backgroundPoor);
    }
}

File: app/src/main/java/gohryt/sapphire/resources/Colors.kt
package gohryt.sapphire.resources

import android.content.Context
import android.os.Build
import androidx.annotation.RequiresApi
import androidx.compose.ui.graphics.Color
import androidx.core.content.ContextCompat
import gohryt.sapphire.R

@RequiresApi(Build.VERSION_CODES.R)
open class Colors(context: Context) {
    var transparent: Color = Color(ContextCompat.getColor(context, R.color.transparent))
    var foregroundMain: Color = Color(ContextCompat.getColor(context, R.color.foregroundMain))
    var foregroundPoor: Color = Color(ContextCompat.getColor(context, R.color.foregroundPoor))
    var foregroundAccent: Color = Color(ContextCompat.getColor(context, R.color.foregroundAccent))
    var backgroundMain: Color = Color(ContextCompat.getColor(context, R.color.backgroundMain))
    var backgroundActive: Color = Color(ContextCompat.getColor(context, R.color.backgroundActive))
    var backgroundPoor: Color = Color(ContextCompat.getColor(context, R.color.backgroundPoor))
    var backgroundAccent: Color = Color(ContextCompat.getColor(context, R.color.backgroundAccent))
}